src/config.ts

```diff
@@ -2,18 +2,21 @@ import { ethers } from 'ethers';
 
 import { RouterConfig } from '@hyperlane-xyz/sdk';
 
-export type Erc20TokenConfig = {
+export type TokenConfig = {
   name: string;
   symbol: string;
   totalSupply: ethers.BigNumberish;
 };
+export type CollateralConfig = {
+  token: string;
+}
 
-export type HypERC20Config = RouterConfig & Erc20TokenConfig;
+export const isCollateralConfig = (config: RouterConfig & TokenConfig | CollateralConfig): config is RouterConfig & CollateralConfig => {
```

Comment:
Is TS treating the config param type as `RouterConfig & (TokenConfig | CollateralConfig)` or `(RouterConfig & TokenConfig) | CollateralConfig` ?
My guess would be latter but code suggest former

Reply:
Also this approach is brittle (breaks if TokenConfig gets a token prop). Probably okay for now but some `_type` prop on which is do discriminated union would be more robust